backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxListenerApi.scala

```diff
@@ -191,4 +137,42 @@ class VeloxListenerApi extends ListenerApi {
   }
 }
 
-object VeloxListenerApi {}
+object VeloxListenerApi {
+  // See org.apache.spark.SparkMasterRegex
+  private val LOCAL_N_REGEX = """local\[([0-9]+|\*)\]""".r
+  private val LOCAL_N_FAILURES_REGEX = """local\[([0-9]+|\*)\s*,\s*([0-9]+)\]""".r
+
+  // TODO: Implement graceful shutdown and remove these flags.
+  //  As spark conf may change when active Spark session is recreated.
+  private val driverInitialized: AtomicBoolean = new AtomicBoolean(false)
+  private val executorInitialized: AtomicBoolean = new AtomicBoolean(false)
+
+  private def inLocalMode(conf: SparkConf): Boolean = {
```

Comment:
nit: This can be can changed to the same code as `Utils.isLocalMaster`.
```
  def isLocalMaster(conf: SparkConf): Boolean = {
    val master = conf.get("spark.master", "")
    master == "local" || master.startsWith("local[")
  }
```

Reply:
Super helpful. Thanks.